pkg/querier/engine/config.go

```diff
@@ -60,5 +64,5 @@ func NewPromQLEngineOptions(cfg Config, activityTracker *activitytracker.Activit
 		NoStepSubqueryIntervalFn: func(int64) int64 {
 			return cfg.DefaultEvaluationInterval.Milliseconds()
 		},
-	}
+	}, cfg.PromQLExperimentalFunctionsEnabled
```

Comment:
Note to reviewers: the reason why I propose to return `cfg.PromQLExperimentalFunctionsEnabled` from `NewPromQLEngineOptions()` even if it's based on the input `cfg` is to have a stronger way to signal the callers to also set the PromQL parser's option to enable the experimental functions.